### PR TITLE
Sl button tests

### PR DESCRIPTION
**Pull request by azizpunjani**
Had to create this PR because I royally messed up the PR https://github.com/softlayer/sl-ember-components/pull/737.

Everything was resolved except for 
>There are not unit tests for currentLabel(), themeClass(), and sizeClass() confirming via _dependentKeys the properties being observed.

Sorry about the inconvenience.

[Review on Reviewable](https://reviewable.io/reviews/softlayer/sl-ember-components/830)

### CODE REVIEW

File: tests/integration/components/sl-button-test.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    assert.ok(
+        noSizeClassByDefault,
+        'No size class is applied by default'
+    );
+});
+
+test( 'Class property is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-button class='testClass'}}
```
Comment (notmessenger) [inline tests/integration/components/sl-button-test.js]: Use double quotes

File: tests/integration/components/sl-button-test.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    assert.strictEqual(
+        this.$( '>:first-child' ).text().trim(),
+        changedPendingLabelText,
+        'Pending label text is set correctly when updated dynamically'
+    );
+});
+
+test( 'Pending class is present when in pending state', function( assert ) {
+    this.render( hbs`
+        {{#sl-button pendingLabel='pending' pending=true}}
```
Comment (notmessenger) [inline tests/integration/components/sl-button-test.js]: Use double quotes

File: tests/integration/components/sl-button-test.js (hunk 1 above)
Comment (azizpunjani) [inline tests/integration/components/sl-button-test.js]: :white_check_mark:

File: tests/integration/components/sl-button-test.js (hunk 2 above)
Comment (azizpunjani) [inline tests/integration/components/sl-button-test.js]: :white_check_mark:

**Comment by notmessenger** [comment]
Linting is failing

**Comment by azizpunjani** [comment]
@notmessenger should be good now, had to merge upstream into my branch.

File: tests/unit/components/sl-button-test.js [hunk 3]
```diff
@@ -155,18 +133,141 @@ test( 'Button supports click event bubbling', function( assert ) {
...
+    assert.ok(
+        spy.calledOnce,
+        'warn() was called when invalid theme was set'
+    );
+
+    utils.warn.restore();
+});
+
+test( 'sizeClass() returns the correct class', function( assert ) {
+    const sizes = Ember.A([
```
Comment (azizpunjani) [inline tests/unit/components/sl-button-test.js]: Use imported Enum.

**Comment by notmessenger** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by azizpunjani** [comment]
Using imported enum now.

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by notmessenger** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by notmessenger** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

<sup>**[tests/unit/components/sl-button-test.js, line 258 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830#-JzzmQa9jNThOsOsxHqX)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/adfcd09610312ae9f9e05e17b860fdc4f9149be9/tests/unit/components/sl-button-test.js#L258)):</sup>
Does the spy need to be restored?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by azizpunjani** [comment]
<sup>**[tests/unit/components/sl-button-test.js, line 258 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830#-JzzmQa9jNThOsOsxHqX)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/adfcd09610312ae9f9e05e17b860fdc4f9149be9/tests/unit/components/sl-button-test.js#L258)):</sup>
Yes, have restored it and pushed up changes.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by notmessenger** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[tests/unit/components/sl-button-test.js, line 258 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830#-JzzmQa9jNThOsOsxHqX)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/adfcd09610312ae9f9e05e17b860fdc4f9149be9/tests/unit/components/sl-button-test.js#L258)):</sup>
I was not referring to the `mockStreamService`, but instead to the `utils.warn()`.  It is restored in another test but not in this one.

I'm assuming `mockStreamService` needs restoring as well though?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by azizpunjani** [comment]
<sup>**[tests/unit/components/sl-button-test.js, line 258 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830#-JzzmQa9jNThOsOsxHqX)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/adfcd09610312ae9f9e05e17b860fdc4f9149be9/tests/unit/components/sl-button-test.js#L258)):</sup>
Ah, yes, both needed restoring. Have pushed up the restore on warn as well.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Comment by notmessenger** [comment]
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/830)

**Pull request merged**